backend/corpora/common/entities/collection.py

```diff
@@ -182,9 +182,35 @@ def publish(self):
         for link in self.links:
             link.collection_visibility = CollectionVisibility.PUBLIC
         for dataset in self.datasets:
-            if dataset.original_id:
-                "skip modified datasets"
-                continue  # TODO: expand to support tombstone and refresh corpora-data-portal/1177
+            ds = Dataset(dataset)
+            ds_original_id = ds.original_id
+            if ds_original_id:
+                original = Dataset.get(self.session, ds_original_id)
+                ds_tombstone = ds.tombstone
+                ds_revision = ds.revision
+                if (public_collection.id, public_collection.visibility) == (
```

Comment:
There is a lot going on in this if statement, what if you pulled it out into a separate check function?

Reply:
that would be a lot of variables going into that functions. How about this 
```
if all([public_collection.id == original.collection_id,
          public_collection.visibility == original.collection_visibility,
          ds_tombstone or ds_revision > original.revision]):
```